bcda/cli.go

```diff
@@ -101,17 +164,27 @@ func importCCLF9(fileMetadata cclfFileMetadata) error {
 	)
 
 	sc := bufio.NewScanner(file)
+	count := 0
+	validator := cclfFileValidator["CCLF9"]
+
 	for sc.Scan() {
 		b := sc.Bytes()
-		if len(bytes.TrimSpace(b)) > 0 {
+		bytelength := len(bytes.TrimSpace(b))
+		if bytelength > 0 && bytelength <= validator.maxRecordLength {
+			count++
+			// currently only errors if there are more records than we expect.
```

Comment:
I think this processing could potentially lead to a file that is only partially ingested.  In other words, we validate in realtime (while ingesting) instead of validating the file up-front (followed by ingesting).  I think we should do the latter, because we don't want to consider ingesting any parts of a file that isn't valid.

Reply:
yep was thinking transactional at the time, but we discussed this in slack so will change.